### PR TITLE
Added example for num and bool to pandas.Series.str.strip

### LINKED ISSUE #44867: BUG: Series.str.strip() removes numeric and bool values instead of only string whitespace
### 

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
import pandas as pd
import numpy as np
s = pd.Series(['hello ', ' goodbye', 1.0, 0.0, 1, 0, True, False, np.nan])
>>> s
0      hello 
1     goodbye
2         1.0
3         0.0
4           1
5           0
6        True
7       False
8         NaN
dtype: object

>>> s.values
array(['hello ', ' goodbye', 1.0, 0.0, 1, 0, True, False, nan],
      dtype=object)

>>> s.str.strip()
0      hello
1    goodbye
2        NaN
3        NaN
4        NaN
5        NaN
6        NaN
7        NaN
8        NaN
dtype: object

>>> s.str.strip().values
array(['hello', 'goodbye', nan, nan, nan, nan, nan, nan, nan],
      dtype=object)
```

### Issue Description

The documentation for `Series.str.strip()` says:

> Strip whitespaces (including newlines) or a set of specified characters from each string in the Series/Index from left and right sides. Equivalent to str.strip().

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.str.strip.html

I assumed that for Series with mixed data types, i.e. that have `dtype=object`, this method would only strip whitespace from string-like entries. Instead, to my surprise, it also replaces numeric and bool values with `NaN`.

I assume this isn't intentional, but if it is intentional, it should be documented.

### Expected Behavior

Numeric and bool values should be left untouched. `Series.str.strip()` should only apply to string values. Here's my naive implementation:

```python
def strip_df(df, colnames=None):
    """Strip leading and trailing whitespace from string-like values of `colnames`
    in DataFrame `df`"""
    df = df.copy() # don't modify in-place
    if colnames is None:
        colnames = df.columns # all columns
    else:
        colnames = np.intersect1d(colnames, df.columns)
    for colname in colnames:
        for i, val in df[colname].iteritems():
            try:
                df.loc[i, colname] = val.strip()
            except AttributeError: # not a string-like value
                pass
    return df
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 66e3805b8cabe977f40c05259cc3fcf7ead5687d
python           : 3.8.10.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-91-lowlatency
Version          : #102-Ubuntu SMP PREEMPT Fri Nov 5 18:18:39 UTC 2021
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.3.5
numpy            : 1.21.4
pytz             : 2021.3
dateutil         : 2.7.3
pip              : 21.3.1
setuptools       : 45.2.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.5.0
html5lib         : 1.0.1
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : 7.30.1
pandas_datareader: None
bs4              : 4.8.2
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.5.1
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by R3dIO**
- [x] closes #44867
- [ ] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/strings/accessor.py [hunk 1]
```diff
@@ -1935,6 +1936,31 @@ def encode(self, encoding, errors="strict"):
     2    Cat
     3    NaN
     dtype: object
+
+    s = pd.Series(['hello ', ' goodbye', 1.0, 0.0, 1, 0, True, False, np.nan])
```
Comment (phofl) [inline pandas/core/strings/accessor.py:1940]: Could you use other numbers int(True) is 1 and int(False) is 0

**Review by phofl — CHANGES_REQUESTED** [review]
The doctests are failing. Could you have a look which one of your changes is causing this?

**Comment by R3dIO** [comment]
Sure I'll have a look

File: pandas/core/strings/accessor.py [hunk 2]
```diff
@@ -1894,7 +1894,8 @@ def encode(self, encoding, errors="strict"):
 
     See Also
     --------
-    Series.str.strip : Remove leading and trailing characters in Series/Index.
+    Series.str.strip : Remove leading and trailing characters in Series/Index,
```
Comment (jreback) [inline pandas/core/strings/accessor.py:1897]: i wouldn't add any of this, instead you can add in a Notes section a line.

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by Anunaya07** [comment]
Hello I am first time contributor. Is this issue still open can I work on it?

**Comment by phofl** [comment]
@R3dIO  are you interested in continuing?

**Comment by R3dIO** [comment]
Sure, let me know how can I help?

**Comment by R3dIO** [comment]
I'll review my PR and fix the changed soon

**Comment by Anunaya07** [comment]
Can I also try too?

**Comment by phofl** [comment]
Please search for another issue labeled as good first issue. We try to keep ownership, so please don’t Open another pr if someone has already started working on it

**Comment by Anunaya07** [comment]
OK. Thank you.

**Comment by mroeschke** [comment]
Thanks for the pull request, but it appears to have gone stale. Please merge in main and address the above comments and we can reopen this PR. Closing.

**Pull request closed without merging**